lib/src/widgets/dashboard_screen/main_drawer.dart

```diff
@@ -16,37 +16,41 @@ class MainDrawer extends StatelessWidget {
                     ? DrawerHeader(child: Container())
                     : StreamBuilder<User>(
                         stream: authenticationBloc.user,
-                        builder: (context, snapshot) =>
-                            UserAccountsDrawerHeader(
-                              decoration: BoxDecoration(
-                                color: SarakaColors.lightRed,
-                              ),
-                              currentAccountPicture: CircleAvatar(
-                                backgroundImage: snapshot.hasData
-                                    ? NetworkImage(snapshot.requireData.imageUrl
-                                        .toString())
-                                    : AssetImage(
-                                        'assets/default_images/user.png'),
-                              ),
-                              accountName: Text(
-                                snapshot.hasData
-                                    ? snapshot.requireData.name
-                                    : 'Loading...',
-                                style: TextStyle(
-                                  fontFamily: SarakaFonts.rubik,
-                                  fontWeight: FontWeight.w500,
-                                ),
+                        builder: (context, snapshot) {
+                          precacheImage(
+                              NetworkImage(
+                                  snapshot.requireData.imageUrl.toString()),
+                              context);
```

Comment:
According to [the document](https://docs.flutter.io/flutter/widgets/precacheImage.html), the ImageProvider you gave `precacheImage()` is supposed to be used for a widget as same. Like this:

```
final image = NetworkImage(snapshot.requireData.imageUrl.toString());

precacheImage(image, context);

return SomeWidget(
  backgroundImage: image,
  ...
);
```

Reply:
I think `precacheImage()` is supposed to be called in `initState` or some lifecycle method else. Because it should be called once for each widget rendering. If it's in `build()` method, it will be called more than once.